src/lib/y2packager/resolvable.rb

```diff
@@ -91,18 +91,28 @@ def initialize(hash)
     def method_missing(method, *args)
       if instance_variable_defined?("@#{method}")
         raise ArgumentError, "Method #{method} does not accept arguments" unless args.empty?
+
         return instance_variable_get("@#{method}")
       end
 
       # load a missing attribute
-      if UNIQUE_ATTRIBUTES.all? { |a| instance_variable_defined?("@#{a}") }
-        load_attribute(method)
-        super unless instance_variable_defined?("@#{method}")
-        raise ArgumentError, "Method #{method} does not accept arguments" unless args.empty?
-        instance_variable_get("@#{method}")
-      else
+      if !UNIQUE_ATTRIBUTES.all? { |a| instance_variable_defined?("@#{a}") }
         raise "Missing attributes for identifying the resolvable."
       end
+
+      load_attribute(method)
+      super unless instance_variable_defined?("@#{method}")
+      raise ArgumentError, "Method #{method} does not accept arguments" unless args.empty?
+
+      instance_variable_get("@#{method}")
+    end
+
+    # defines for dynamic methods also respond_to?
+    def respond_to_missing?(method, _private)
+      return true if instance_variable_defined?("@#{method}")
```

Comment:
This whole function body can be replaced by
```ruby
instance_variable_defined?("@#{method}") || UNIQUE_ATTRIBUTES.include?(method.to_sym)
```

Reply:
yes, for me the one with early exit is more readable as it follows structure of method_missing, but I can change it.

Reply:
ok, I see